server/repository/src/migrations/v2_04_00/add_reason_option_table.rs

```diff
@@ -0,0 +1,45 @@
+use crate::migrations::*;
+
+pub(crate) struct Migrate;
+
+impl MigrationFragment for Migrate {
+    fn identifier(&self) -> &'static str {
+        "add_reason_option_table"
+    }
+
+    fn migrate(&self, connection: &StorageConnection) -> anyhow::Result<()> {
+        if cfg!(feature = "postgres") {
+            sql!(
+                connection,
+                r#"
+                CREATE TYPE reason_option_type AS ENUM (
+                'POSITIVE_INVENTORY_ADJUSTMENT,
+                'NEGATIVE_INVENTORY_ADJUSTMENT',
+                'RETURN_REASON',
+                'REQUISITION_LINE_VARIANCE
+                );
+            "#
+            )?;
+        }
+
+        const OPTION_TYPE_ENUM: &str = if cfg!(feature = "postgres") {
+            "reason_option_type"
+        } else {
+            "TEXT"
+        };
+
+        sql!(
+            connection,
+            r#"
+                CREATE TABLE reason_option (
```

Comment:
Renamed since Rust has optional.. I guess this could just be reason and the column can be renamed... Unsure what the column would be called tho 🤔  suggestions?